MetaMask/metamask-extension
# Version v10.26.0 lavamoat patches

## Explanation

The v10.26.0 was not working on chrome v78. We were seeing the following errors, each of which have been fixed in this PR. Some of the fixes are via patches only. All of the fixes in this PR can be temporary, and don't _need_ to make it to the develop branch, as lavamoat is currently working on that branch.

The errors and there fixes are:

- `app/scripts/use-snow.js` was using the optional chaining operator, which is not supported in v78
- a scuttling exception was needed for `location`, as this error was being hit https://github.com/LavaMoat/LavaMoat/blob/main/packages/core/src/kernelCoreTemplate.js#L131-L136
- patches were needed for the `kernalCoreTemplate.js` in `lavamoat-core`, as those are being used to generate the `runtime-lavamoat.js` file via `lavapack.buildRuntime` in `development/build/index.js`, and `runtime-lavamoat.js` has optional chaining operators
- a patch was made to `snow.prod.js` in `@lavamoat/snow`, which was trying to find descriptors for methods that were not introduced until later versions, notably `replaceChildren`. The code that had to be patched is here https://github.com/LavaMoat/snow/blob/1e04737d7ea50f25af24110dc951af05a26ebabf/snow.js#L411-L415



## Manual Testing Steps

Install chrome v78 locally, build this branch with `yarn dist`, install that build in chrome v78. You should be able to use MetaMask normally

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
